RiotSwiftUI/Modules/UserSessions/UserSessionOverview/Service/MatrixSDK/UserSessionOverviewService.swift

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright 2022 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Combine
+import MatrixSDK
+
+class UserSessionOverviewService: UserSessionOverviewServiceProtocol {
+
+    // MARK: - Members
+    
+    private(set) var pusherEnabledSubject: CurrentValueSubject<Bool?, Never>
+    private(set) var remotelyTogglingPushersAvailableSubject: CurrentValueSubject<Bool, Never>
+
+    // MARK: - Private
+    
+    private let session: MXSession
+    private let sessionInfo: UserSessionInfo
+    private var pusher: MXPusher?
+    
+    // MARK: - Setup
+    
+    init(session: MXSession, sessionInfo: UserSessionInfo) {
+        self.session = session
+        self.sessionInfo = sessionInfo
+        self.pusherEnabledSubject = CurrentValueSubject(nil)
+        self.remotelyTogglingPushersAvailableSubject = CurrentValueSubject(false)
+
+        checkServerVersions { [weak self] in
+            self?.checkPusher()
+        }
+    }
+    
+    // MARK: - UserSessionOverviewServiceProtocol
+    
+    func togglePushNotifications() {
+        guard let pusher = pusher, let enabled = pusher.enabled?.boolValue, self.remotelyTogglingPushersAvailableSubject.value else {
+            return
+        }
+        
+        let data = pusher.data.jsonDictionary() as? [String: Any] ?? [:]
+        
+        self.session.matrixRestClient .setPusher(pushKey: pusher.pushkey, kind: MXPusherKind.from(value: pusher.kind), appId: pusher.appId, appDisplayName:pusher.appDisplayName, deviceDisplayName: pusher.deviceDisplayName, profileTag: pusher.profileTag ?? "", lang: pusher.lang, data: data, append: false, enabled: !enabled) { [weak self] response in
```

Comment:
```suggestion
        self.session.matrixRestClient.setPusher(pushKey: pusher.pushkey, kind: MXPusherKind.from(value: pusher.kind), appId: pusher.appId, appDisplayName:pusher.appDisplayName, deviceDisplayName: pusher.deviceDisplayName, profileTag: pusher.profileTag ?? "", lang: pusher.lang, data: data, append: false, enabled: !enabled) { [weak self] response in
```
(Should probably be split onto multiple lines too)